server/src/zmq_worker.rs

```diff
@@ -304,38 +304,41 @@ async fn handle_incoming_request(
                         .context("ZmqWorker error while forwarding request to load balancer.")
                 }
                 _ => {
-                    error!(
-                        "ZmqWorker received a request with api {:?} while I can only handle \
-                    Status/Metadatas/PtPlanner/PlacesNearby/NextDepartures/NextArrivals api.",
-                        requested_api
-                    );
-                    Ok(())
+                    let err = format_err!("ZmqWorker received a request with api {requested_api:?} while I can only handle \
+                    Status/Metadatas/PtPlanner/PlacesNearby/NextDepartures/NextArrivals api.");
+                    error!("{:?}", err);
+                    // let's send back a response to our zmq client that we received an invalid query
+                    send_error_response(zmq_socket, request_message.client_id, &err).await
                 }
             }
         }
         Err(err) => {
-            let err_str = format!("Could not decode zmq message into protobuf. {:?}", err);
-            error!("{}", err_str);
-
+            let err = format_err!(err).context("Could not decode zmq message into protobuf.");
+            error!("{:?}", err);
             // let's send back a response to our zmq client that we received an invalid protobuf
-            let response_proto = make_error_response(&format_err!("{}", err_str));
-            let response_message = ResponseMessage {
-                client_id,
-                payload: response_proto,
-            };
-            send_response_to_zmq(zmq_socket, response_message).await
+            send_error_response(zmq_socket, client_id, &err).await
         }
     }
 }
 
-fn make_error_response(error: &Error) -> navitia_proto::Response {
+async fn send_error_response(
+    zmq_socket: &mut tmq::router::Router,
+    client_id: tmq::Multipart,
+    error: &Error,
+) -> Result<(), Error> {
     let mut proto_response = navitia_proto::Response::default();
     proto_response.set_response_type(navitia_proto::ResponseType::NoSolution);
     let mut proto_error = navitia_proto::Error::default();
     proto_error.set_id(navitia_proto::error::ErrorId::InternalError);
     proto_error.message = Some(format!("{}", error));
```

Comment:
I know it's old code, but how about [`.to_string()` instead](https://stackoverflow.com/a/27770058/7447059)? The use of `format!()` macro in this case is convoluted...

```suggestion
    proto_error.message = Some(error.to_string());
```

Reply:
:+1: